CloudFlareDDNS-CLI/Program.cs

```diff
@@ -84,14 +79,57 @@ static void Main(string[] args)
                 {
                     user = user,
                     record = record,
-                    zone_id = config_root.SelectSingleNode("/config/zone_id").InnerText
+                    zone_id = config_root.SelectSingleNode("/UserConfigurationData/zone_id").InnerText
                 };
                 ConsoleLog.Info("Site | zone_id | " + request.zone_id);
-                //API列表读取
-                XmlNodeList api_node_list = config_root.SelectNodes("/config/ipapis/api");
+            }
+            catch (NullReferenceException e)
+            {
+                ConsoleLog.Error("Config file missing some information! Details below.");
+                ConsoleLog.Info(e.Message);
+                ConsoleLog.Log("Exit.");
+                return;
+            }
+            catch (FormatException e)
+            {
+                ConsoleLog.Error("Config file's format is wrong! Details below.");
+                ConsoleLog.Info(e.Message);
+                ConsoleLog.Log("Exit.");
+                return;
+            }
+            ConsoleLog.Succeed("Successfully read configuration.");
+            #endregion
+
+            #region 打开API配置文件
+            XmlDocument api_xml = new XmlDocument();
+            try
+            {
+                api_xml.Load("config_api.xml");
+            }
+            catch (FileNotFoundException e)
+            {
+                ConsoleLog.Error("Config file not found! Details below.");
+                ConsoleLog.Info(e.Message);
+                ConsoleLog.Log("Exit.");
+                return;
+            }
+            catch (XmlException e)
+            {
+                ConsoleLog.Error("Config file data invalid.");
+                ConsoleLog.Info(e.Message);
+                ConsoleLog.Log("Exit.");
+                return;
+            }
+            XmlElement api_root = api_xml.DocumentElement;
+            ConsoleLog.Succeed("API Configuration File opened.");
```

Comment:
I wonder what is the necessity to open a new configuration file. If need to separate user and program information, just separate user and inner configuration, so why should we pull the APIs out ? If users are authorized to edit the APIs, they should also be authorized to modify the information of their site. (Just wondering)

Reply:
Because I haven't found any API that can write the `List<>` attributes in the class to the `XML` file. Also, a reason for doing this is that it is the most convenient way to implement and code-efficient (laugh)/.

Reply:
It's just … so ugly. This is the first time that I have seen a configuration file being separated like this. If config.xml and config_api.xml were merged, will the "Deserializer/Serializer" go wrong ? Maybe we should find some better solution in the future. (I will create an issue.) I recommend merging these configurations files into one file. (Otherwise it's just too ugly.)  😢

Reply:
Honestly, they are indeed ugly codes and it is necessary to optimize them in the future. Maybe we should directly handle with the file instead of using `Serializer` / `Deserializer`, or somewhat reorganize our data structure.